src/proxy/moralis.rs

```diff
@@ -0,0 +1,310 @@
+use std::str::FromStr;
+
+pub(crate) async fn proxy_moralis(
+    cfg: &AppConfig,
+    mut req: Request<Body>,
+    remote_addr: &SocketAddr,
+    signed_message: SignedMessage,
+    x_forwarded_for: HeaderValue,
+    proxy_route: &ProxyRoute,
+) -> GenericResult<Response<Body>> {
+    if proxy_route.authorized {
+        if let Err(e) = insert_jwt_to_http_header(cfg, req.headers_mut()).await {
+            log::error!(
+                "{}",
+                log_format!(
+                    remote_addr.ip(),
+                    signed_message.address,
+                    req.uri(),
+                    "Error inserting JWT into HTTP header: {}, returning 500.",
+                    e
+                )
+            );
+            return response_by_status(StatusCode::INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    let original_req_uri = req.uri().clone();
+
+    if let Err(e) = modify_request_uri(&mut req, proxy_route).await {
+        log::error!(
+            "{}",
+            log_format!(
+                remote_addr.ip(),
+                signed_message.address,
+                original_req_uri,
+                "Error modifying request Uri: {}, returning 500.",
+                e
+            )
+        );
+        return response_by_status(StatusCode::INTERNAL_SERVER_ERROR);
+    }
+
+    remove_hop_by_hop_headers(&mut req, &[header::CONTENT_LENGTH])?;
+
+    req.headers_mut()
+        .insert(HeaderName::from_static(X_FORWARDED_FOR), x_forwarded_for);
+    req.headers_mut()
+        .insert(header::ACCEPT, APPLICATION_JSON.parse()?);
+
+    let https = HttpsConnector::new();
+    let client = hyper::Client::builder().build(https);
+
+    let target_uri = req.uri().clone();
+    let res = match client.request(req).await {
+        Ok(t) => t,
+        Err(e) => {
+            log::warn!(
+                "{}",
+                log_format!(
+                    remote_addr.ip(),
+                    signed_message.address,
+                    original_req_uri,
+                    "Couldn't reach {}: {}. Returning 503.",
+                    target_uri,
+                    e
+                )
+            );
+            return response_by_status(StatusCode::SERVICE_UNAVAILABLE);
+        }
+    };
+
+    Ok(res)
+}
+
+/// Modifies the URI of an HTTP request by replacing its base URI with the outbound URI specified in `ProxyRoute`,
+/// while incorporating the path and query parameters from the original request URI. Additionally, this function
+/// removes the first path segment from the original URI.
+async fn modify_request_uri(
+    req: &mut Request<Body>,
+    proxy_route: &ProxyRoute,
+) -> GenericResult<()> {
+    let proxy_base_uri = proxy_route.outbound_route.parse::<Uri>()?;
+
+    let req_uri = req.uri().clone();
+
+    // Remove the first path segment
+    let mut path_segments: Vec<&str> = req_uri
+        .path()
+        .split('/')
+        .filter(|s| !s.is_empty())
+        .collect();
```

Comment:
I suppose this is the same reason why we skip the first segment during inbound route parsing.

Reply:
Yes, in `modify_request_uri` we should remove inbound_route from req uri and change base url

Reply:
done https://github.com/KomodoPlatform/komodo-defi-proxy/pull/22/commits/e81cfb91ad505da4b36d23e55220d26850d904b8